### ISSUE TITLE
NRE in OpenApiAnyFactory.TryCreateFor if value is null

### ISSUE THREAD

**Issue by Hoffs** [domaindrivendev/Swashbuckle.AspNetCore#1407]
https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/a025be2cae11683ef473275041bd5090be6e3ce1/src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/OpenApiAnyFactory.cs#L9

So if you call `OpenApiAnyFactory.TryCreateFor(parameter.Schema, description.DefaultValue, out var openApiAny)` where `parameter` is some `OpenApiParameter` and `description` is `ApiParameterDescription` for that parameter. 
So if it's schema has type of `string` it goes to L35 and then NRE's because of `ToString`
https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/a025be2cae11683ef473275041bd5090be6e3ce1/src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/OpenApiAnyFactory.cs#L35

But the core issue is that in my opinion calling this 
`TryX` method you should not get exception if `value` is null, but rather return false.

Full snippet of operation filter with this usage:
```csharp
public void Apply(OpenApiOperation operation, OperationFilterContext context)
        {
            var apiDescription = context.ApiDescription;
            var apiVersion = apiDescription.GetApiVersion();

            #pragma warning disable S3265 // bitfields
            var model = apiDescription.ActionDescriptor.GetApiVersionModel(Explicit | Implicit);
            #pragma warning restore S3265

            operation.Deprecated = model.DeprecatedApiVersions.Contains(apiVersion);

            if (operation.Parameters == null)
            {
                return;
            }

            foreach (var parameter in operation.Parameters)
            {
                var description = apiDescription.ParameterDescriptions
                                                .First(p => p.Name == parameter.Name);

                if (parameter.Description == null)
                {
                    parameter.Description = description.ModelMetadata?.Description;
                }

                if (parameter.Schema.Default == null &&
                    OpenApiAnyFactory.TryCreateFor(parameter.Schema, description.DefaultValue, out var openApiAny))
                {
                    parameter.Schema.Default = openApiAny;
                }

                parameter.Required |= description.IsRequired;
            }
        }
```

**Comment by Hoffs** [comment]
Same as #1159 I believe

**Comment by domaindrivendev** [comment]
Fixed by 5a4b35f0aa7a6a41077d1e3ec2123299b8bff0ad
